persim/visuals.py

```diff
@@ -280,3 +280,178 @@ def wasserstein_matching(I1, I2, matchidx, palette=None, labels=["dgm1", "dgm2"]
+                typically the output of ripser(nodes)['dgms']
+
+        examples:
+        ===========
+
+        n = 300
+        t = np.linspace(0, 2 * np.pi, n)
+        noise = np.random.normal(0, 0.1, size=n)
+
+        data = np.vstack([((3+d) * np.cos(t[i]+d), (3+d) * np.sin(t[i]+d)) for i, d in enumerate(noise)])
+        diagrams = ripser(data)
+
+        bc = Barcode(diagrams['dgms'])
+        bc.plot_barcode()
+        '''
+        if not isinstance(diagrams, list):
+            diagrams = [diagrams]
+
+        if len(diagrams) == 2:
+            self.plot_barcode = self._plot_H0_H1
+
+        else:
+            self.plot_barcode = self._plot_Hn
+
+        self.diagrams = diagrams
+
+    def _plot_H0_H1(self, **kwargs):
+        '''
+        parameters:
+        ===========
+            figsize: tuple
+                figure size, default=(6,6)
+
+            show: boolean
+                show the figure via plt.show()
+
+            export_png: boolean
+                write image to png data, returned as io.BytesIO() instance, default=False
+                
+            **kwargs: artist paramters for the barcodes, defaults:
+                c='grey'
+                linestyle='-'
+                linewidth=0.5
+                dpi=100 (for png export)
+
+        returns:
+        ===========
+            list of png exports or []
+        '''
+        import io
+
+        fsize = kwargs.get('figsize', (6, 6))
+        show = kwargs.get('show', True)
+        export = kwargs.get('export_png', False)
+        dpi = kwargs.get('dpi', 100)
+
+        out = []
+
+        fig, ax = plt.subplots(2, 1, figsize=fsize)
+
+        for dim, diagram in enumerate(self.diagrams):
+            self._plot_many_bars(dim, diagram, dim, ax, **kwargs)
+
+        if export:
+            fp = io.BytesIO()
+            plt.savefig(fp, dpi=dpi)
+            fp.seek(0)
+
+            out += [fp]
+
+        if show:
+            plt.show()
+
+        return out
+
+    def _plot_Hn(self, **kwargs):
+        '''
+        parameters:
+        ===========
+            figsize: tuple
+                figure size, default=(6,6)
+
+            show: boolean
+                show the figure via plt.show()
+
+            export_png: boolean
+                write image to png data, returned as io.BytesIO() instance, default=False
+                
+            **kwargs: artist paramters for the barcodes, defaults:
+                c='grey'
+                linestyle='-'
+                linewidth=0.5
+                dpi=100 (for png export)
+
+        returns:
+        ===========
+            list of png exports or []
+        '''
+        import io
+        
+        fsize = kwargs.get('figsize', (6, 4))
+        show = kwargs.get('show', True)
+        export = kwargs.get('export_png', False)
+        dpi = kwargs.get('dpi', 100)
+
+        out = []
+
+        for dim, diagram in enumerate(self.diagrams):
+            fig, ax = plt.subplots(1, 1, figsize=fsize)
```

Comment:
could you let fig and ax be passed in? this would follow the convention we've used elsewhere in the library.

take `ax` as an arg with default `None`, then set `ax = ax or plt.gca()` at the top of the function.

This will give users more ability to add the plots to their own subplot arrangement.

Reply:
Also, it doesn't look like `fig` is used anyways.

Reply:
`fig` is indeed unused, so removed. Given the way the wrapper class has been set up however, I don't see a clean way to pass `ax` around.